yt/visualization/plot_window.py

```diff
@@ -1118,6 +1142,14 @@ def _setup_plots(self):
                 ia = ImageArray(ia)
             else:
                 ia = image
+
+            swap_axes = self._swap_axes
+            aspect = self.aspect
+            if swap_axes:
+                extent = [extent[i] for i in (2, 3, 0, 1)]
```

Comment:
took me a minute to read but I think this is a place where you'd want to use a `collections.deque` just for the sake of the `deque.rotate` method (provided that's actually what's happening here)
https://docs.python.org/3/library/collections.html#collections.deque.rotate

Reply:
cool, I'll check out deque. The simpler to read but longer code I had initially was just `extent = (extent[2], extent[3], extent[0], extent[1])`.

Reply:
So in the latest batch of commits, I just created a `_swap_axes_extents` function that just does this very explicitly. I didn't use deque cause I thought it was overkill for such a simple thing. But I don't feel strongly about this.

Reply:
I forgot to reply here: your solution is great, I agree that using a duque would be an overkill